CODE_STYLE.md

```diff
@@ -0,0 +1,260 @@
+      datasetSchema {
+        id
+        name
+      }
+    }
+  }
+`;
+```
+
+</td><td>
+
+```TypeScript
+export const IMPORT_DATASET = gql`
+  mutation ImportDataset($file: Upload!, $sceneId: ID!, $datasetSchemaId: ID) {
+    importDataset(input: { file: $file, sceneId: $sceneId, datasetSchemaId: $datasetSchemaId }) {
+      datasetSchema {
+        id
+        name
+      }
+    }
+  }
+`;
+```
+
+</td></tr>
+</tbody></table>
+
+### Functions
+
+**Naming**
+
+We have two styles when naming functions: `onXXXX` and `handleXXXX`. Note both are camelCase. We use `onXXXX` for functional props, and we use `handleXXXX` when we are locally declaring the function.
+
+**Word order**
+
+Should be `nounVerb`. Combining this with deliberate prop or export order keeps things very easy to read.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```TypeScript
+onRemoveAsset
+onHandleAddDataset
+OnUpdatetheme
+handleSyncDataset
+updateProject
+fetchuser
+```
+
+</td><td>
+
+```TypeScript
+onAssetAdd
+onAssetRemove
+onAssetUpdate
+
+handleProjectAdd
+handleProjectRemove
+handleProjectUpdate
+```
+
+</td></tr>
+</tbody></table>
+
+
+### Variables
+
+**Naming**
+
+Descriptive over short. Context matters, of course, so generic code can and should have `name` instead of `teamName`. But typically `teamName` is preferable over `name`.
+
+### Commenting
+
+Follow Clean Code unless absolutely necessary. Which, in a nutshell, is "If you need a comment, either your function or variable isn't named appropriately, or your function is doing too much and should be broken down into simpler functions".
+
+### Atomic design
+
+Re:Earth is broken up into 4 component categories: `Pages`, `Organisms`, `Molecules` and `Atoms`. See the component directory's [README](https://github.com/reearth/reearth-web/blob/main/src/components/README.md) for details specific to Re:Earth.
+
+### How to write paths
+
+Unless heavily nested and used only in its own or its parent directory, start from the `@reearth/` alias for the src directory.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```TypeScript
+import CoolComponent from "../../../../CoolComponent;
+```
+
+</td><td>
+
+```TypeScript
+import CoolComponent from "@reearth/components/atoms/CoolComponent;
+```
+
+</td></tr>
+</tbody></table>
+
+### Spacing and order
+
+*NOTE: Often ESlint will catch spacing issues on save, but not always. Because of this, be mindful of your spacing as it affects the file's readability immensely. For order, ESlint will only fix import order.*
+
+Since file content varies in size, complexity, whether it is a component or utility, hooks, etc., it is hard to set specific rules for spacing and order. We only ask that you be deliberate (and take time refactoring before opening a PR) and be open to suggestions on spacing or order during PR review to keep code as readable and accessible to new eyes as possible.
+
+### Destructuring
```

Comment:
This rule can be detected by ESLint, so it would be better to set up ESLint than to write it here, so everyone will abide by this rule.

https://eslint.org/docs/rules/prefer-destructuring

Reply:
Ok, I'll look into that!